src/scripts/main.js

```diff
@@ -1,5 +1,38 @@
 'use strict';
 
+const spider = document.querySelector('.spider');
+const wall = document.querySelector('.wall');
+// we create values from elements before point
+// when we set handler and use those as we need
+
 document.addEventListener('click', (e) => {
-  // write code here
+  const field = e.target.closest('.wall');
+
+  if (!field) {
+    return; // we get rid of options when "click" happens outside the wall
+  }
+
+  const wallRect = wall.getBoundingClientRect();
+  const spiderRect = spider.getBoundingClientRect();
+  const widthSpider = spider.offsetWidth; // the spider is square
+  const border = wall.clientTop;
+
+  let x = e.clientX - widthSpider / 2 - border - wallRect.left;
+  let y = e.clientY - widthSpider / 2 - border - wallRect.top;
+
+  // here we math how many 'px' the spider need to move on
+  // the wall in vertical and horizontal parameters relative himself.
+  // From coordinates of "click" we subtract half width spider,
+  // border of wall, position wall in our window and
+  // get a numbers those we need
+
+  x = Math.max(0, Math.min(x, wall.clientWidth - spiderRect.width));
```

Comment:
The calculations for 'x' and 'y' here are a bit hard to understand. Consider adding a comment explaining why you're using `Math.max` and `Math.min` to calculate the position.